src/core/qgstaskmanager.cpp

```diff
@@ -142,26 +144,19 @@ QList<QgsMapLayer *> QgsTask::dependentLayers() const
   return _qgis_listQPointerToRaw( mDependentLayers );
 }
 
-bool QgsTask::waitForFinished( int timeout )
+bool QgsTask::waitForFinished( unsigned long timeout )
 {
-  QEventLoop loop;
   bool rv = true;
-
-  connect( this, &QgsTask::taskCompleted, &loop, &QEventLoop::quit );
-  connect( this, &QgsTask::taskTerminated, &loop, &QEventLoop::quit );
-  QTimer timer;
-
-  if ( timeout != -1 )
+  if ( mOverallStatus == Complete || mOverallStatus == Terminated )
   {
-    timer.start( timeout );
-    connect( &timer, &QTimer::timeout, [&rv]() { rv = false; } );
-    connect( &timer, &QTimer::timeout, &loop, &QEventLoop::quit );
+    rv = true;
+  }
+  else
+  {
+    if ( timeout == -1 )
```

Comment:
If `timeout` is an unsigned long, I suppose that it will never be equal to `-1`.

Reply:
Oops, I just pushed an updated version but apparently only changed the signature. Good catch.

Reply:
Ah, QtCreator had not saved the file...